compiler/optimizer/LoopVersioner.cpp

```diff
@@ -4775,6 +4815,24 @@ void TR_LoopVersioner::versionNaturalLoop(TR_RegionStructure *whileLoop, List<TR
       properRegion->removeExternalEdgeTo(predBlock->getStructureOf(), succBlock->getStructureOf()->getNumber());
       }
 
+   if (seenOSRGuards && safeToRemoveOSRGuards)
```

Comment:
Can probably use a comment as well to describe what is going to happen and why like most of the other big blocks in this area.